extensions/azcli/src/az.ts

```diff
@@ -285,36 +267,70 @@ async function executeAzCommand(command: string, args: string[], additionalEnvVa
 	return executeCommand(command, args, additionalEnvVars);
 }
 
-// Commented the install/update prompts out until DoNotAskAgain is implemented
-// async function setConfig(key: string, value: string): Promise<void> {
-// 	const config = vscode.workspace.getConfiguration(azConfigSection);
-// 	await config.update(key, value, vscode.ConfigurationTarget.Global);
-// }
+function getConfig(key: string): AzDeployOption | undefined {
+	const config = vscode.workspace.getConfiguration(azConfigSection);
+	const value = <AzDeployOption>config.get<AzDeployOption>(key);
+	return value;
+}
+
+async function setConfig(key: string, value: string): Promise<void> {
+	const config = vscode.workspace.getConfiguration(azConfigSection);
+	await config.update(key, value, vscode.ConfigurationTarget.Global);
+}
+
 
 /**
- * Find user's local Azure CLI. Execute az --version and parse out the version number.
- * If an update is needed, prompt the user to update via link. Return the AzTool.
- * Currently commented out because Don't Prompt Again is not properly implemented.
+ * Finds and returns the user's locally installed Azure CLI tool. Checks to see if arcdata extension is
+ * installed and updated. If no Azure CLI is found, error will be thrown.
  */
-async function findSpecificAz(): Promise<IAzTool> {
-	const path = await ((process.platform === 'win32') ? searchForCmd('az.cmd') : searchForCmd('az'));
+export async function findAzAndCheckArcdata(): Promise<IAzTool> {
+	let response: string | undefined;
+	const responses = [loc.askLater, loc.doNotAskAgain];
+	let path = undefined;
+
+	try {
+		path = await ((process.platform === 'win32') ? searchForCmd('az.cmd') : searchForCmd('az'));
+	} catch (err) {
+		// If no Azure CLI was found, prompt to install
+		const azInstallNeededConfig = <AzDeployOption>getConfig(azCliInstallKey);
+		if (azInstallNeededConfig !== AzDeployOption.dontPrompt) {
+			response = await vscode.window.showErrorMessage(loc.noAzureCLI, ...responses);
+			if (response === loc.doNotAskAgain) {
+				await setConfig(azCliInstallKey, AzDeployOption.dontPrompt);
+			}
+		}
+		throw err;
+	}
+
 	const versionOutput = await executeAzCommand(`"${path}"`, ['--version']);
-	const version = parseArcExtensionVersion(versionOutput.stdout);
-	const semVersion = new SemVer(version);
-	//let response: string | undefined;
+	const azVersion = parseVersion(versionOutput.stdout);
```

Comment:
If this doesn't parse correctly it doesn't look like you handle that - we should at least log a warning or something